app/code/core/Mage/Admin/Model/Redirectpolicy.php

```diff
@@ -39,7 +39,7 @@ public function getRedirectUrl(
         ?Zend_Controller_Request_Http $request = null,
         $alternativeUrl = null
     ) {
-        if (empty($request)) {
+        if (!$request instanceof Zend_Controller_Request_Http) {
```

Comment:
Is this better:
```suggestion
        if (is_null(request)) {
```

given line 39:
```
        ?Zend_Controller_Request_Http $request = null,
```